scaffolding-chef-inspec/lib/scaffolding.ps1

```diff
@@ -0,0 +1,176 @@
+`$env:PATH = "{{pkgPathFor "stuartpreston/inspec"}}/bin;`$env:PATH"
+`$env:INSPEC_CONFIG_DIR="{{pkg.svc_var_path}}"
+
+
+`$env:CFG_SPLAY_FIRST_RUN="{{cfg.splay_first_run}}"
+if(!`$env:CFG_SPLAY_FIRST_RUN) {
+    `$env:CFG_SPLAY_FIRST_RUN = "0"
+}
+`$env:CFG_INTERVAL="{{cfg.interval}}"
+if(!`$env:CFG_INTERVAL){
+    
+}
+`$env:CFG_SPLAY="{{cfg.splay}}"
+if(!`$env:CFG_SPLAY){
+    `$env:CFG_SPLAY = "1800"
+}
+`$env:CFG_LOG_LEVEL="{{cfg.log_level}}"
+if (!`$env:CFG_LOG_LEVEL){
+    `$env:CFG_LOG_LEVEL = "warn"
+}
+`$env:CFG_CHEF_LICENSE="{{cfg.chef_license.acceptance}}"
+if(!`$env:CFG_CHEF_LICENSE){
+    `$env:CFG_CHEF_LICENSE = "undefined"
+}
+`$CONFIG="{{pkg.svc_config_path}}/inspec_exec_config.json"
+`$PROFILE_PATH="{{pkg.path}}/{{pkg.name}}-{{pkg.version}}.tar.gz"
+
+function Invoke-Inspec {
+    inspec exec `$PROFILE_PATH --config `$CONFIG --log-level `$CFG_LOG_LEVEL
+}
+
+`$SPLAY_DURATION = Get-Random -InputObject (0..`$env:CFG_SPLAY) -Count 1
+
+`$SPLAY_FIRST_RUN_DURATION = Get-Random -InputObject (0..`$env:CFG_SPLAY_FIRST_RUN) -Count 1
+
+
+Start-Sleep -Seconds `$SPLAY_FIRST_RUN_DURATION
+Invoke-Inspec
+
+while(`$true){
+  `$SLEEP_TIME = `$SPLAY_DURATION + `$env:CFG_INTERVAL
+  Write-Host "InSpec is sleeping for `$SLEEP_TIME seconds"
+  Start-Sleep -Seconds `$SPLAY_DURATION
+  Start-Sleep -Seconds `$env:CFG_INTERVAL
+  Invoke-Inspec
+}
+"@
+}
+
+function Invoke-DefaultBuild {
+    Copy-Item $PLAN_CONTEXT/../* $HAB_CACHE_SRC_PATH/$pkg_dirname -Recurse -Force
+    inspec archive "$HAB_CACHE_SRC_PATH/$pkg_dirname" `
+                   --overwrite `
+                   -o "$HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name-$pkg_version.tar.gz"
+}
+
+function Invoke-DefaultInstall {
+    Copy-Item "$HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name-$pkg_version.tar.gz" "$pkg_prefix"
+    New-Item -Type Directory $pkg_prefix/config
+
+    Write-BuildLine "Generating config/inspec_exec_config.json"
+    Add-Content -Path "$pkg_prefix/config/inspec_exec_config.json" -Value @"
+{
+    "target_id": "{{ sys.member_id }}",
+    "reporter": {
+        "cli": {
+          "stdout": {{cfg.report_to_stdout}}
+        },
+        "json": {
+          "file": "{{pkg.svc_path}}/logs/inspec_last_run.json"
+        }{{#if cfg.automate.enable ~}},
+        "automate" : {
+          "url": "{{cfg.automate.server_url}}/data-collector/v0/",
+          "token": "{{cfg.automate.token}}",
+          "node_name": "{{ sys.hostname }}",
+          "verify_ssl": false
+        }{{/if ~}}
+    }
+    {{#if cfg.automate.enable }},
+    "compliance": {
+      "server" : "{{cfg.automate.server_url}}",
+      "token" : "{{cfg.automate.token}}",
+      "user" : "{{cfg.automate.user}}",
+      "insecure" : true,
+      "ent" : "automate"
+    }{{/if }}
+}
+"@
+
+    Write-BuildLine "Generating config/cli_only.json"
+    Add-Content -Path "$pkg_prefix/config/cli_only.json" -Value @"
+{
+    "reporter": {
+        "cli" : {
+            "stdout" : true
+        }
+    }
+}
+"@
+
```

Comment:
Can you switch the order of these two files ^^^^ so they match the same order as the linux scaffolding?

Reply:
absolutely